tests/test_main_sync_error_handling.py

```
@@ -0,0 +1,292 @@
+
+def test_sync_timeout_logs_warning_and_retries():
+    """TimeoutError from sync_task.result() logs warning and retries."""
+    config = {
+        "matrix_rooms": [{"id": "!room:matrix.org"}],
+        "matrix": {"homeserver": "https://matrix.org"},
+        "meshtastic": {"connection_type": CONNECTION_TYPE_SERIAL},
+    }
+
+    call_count = [0]
+
+    async def run_test():
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `run_test`

(ANN202)